src/modules/sharing/sharing.service.ts

```diff
@@ -286,6 +305,89 @@ export class SharingService {
     return response as PublicSharingInfo;
   }
 
+  async getPublicSharingItemInfo(id: Sharing['id']): Promise<SharingItemInfo> {
+    const sharing = await this.sharingRepository.findOneSharing({
+      id,
+    });
+
+    if (!sharing) {
+      throw new NotFoundException();
+    }
+
+    if (!sharing.isPublic()) {
+      throw new ForbiddenException();
+    }
+
+    let item: Item;
+    if (sharing.itemType === 'file') {
+      item = await this.fileUsecases.getByUuid(sharing.itemId);
+      if (item.isDeleted()) {
+        throw new NotFoundException();
+      }
+    } else {
+      item = await this.folderUsecases.getByUuid(sharing.itemId);
+      if (item.isRemoved()) {
+        throw new NotFoundException();
+      }
+    }
+
+    if (!item.plainName) {
+      if (item instanceof File) {
+        item.plainName = this.fileUsecases.decrypFileName(item).plainName;
+      } else {
+        item.plainName = this.folderUsecases.decryptFolderName(item).plainName;
+      }
+    }
+    return {
+      plainName: item.plainName,
+      type: item.type,
+      size: item.size,
+    };
+  }
+
+  async setSharingPassword(
+    user: User,
+    id: Sharing['id'],
+    encryptedPassword: string,
+  ): Promise<Sharing> {
+    const sharing = await this.sharingRepository.findOneSharing({
+      id,
+    });
+
+    if (!sharing) {
+      throw new NotFoundException();
+    }
+
+    if (!sharing.isPublic() || !sharing.isOwnedBy(user)) {
```

Comment:
If it is not public and someone attempts to set a password, instead of Forbidden, I suggest returning a ConflictException / BadRequestException as it is more semantic/explanatory.